collector/pkg/component/analyzer/network/protocol/rocketmq/rocketmq_request.go

```diff
@@ -0,0 +1,58 @@
+package rocketmq
+
+import (
+	"encoding/json"
+	"github.com/Kindling-project/kindling/collector/pkg/component/analyzer/network/protocol"
+	"github.com/Kindling-project/kindling/collector/pkg/model/constlabels"
+)
+
+func fastfailRocketMQRequest() protocol.FastFailFn {
+	return func(message *protocol.PayloadMessage) bool {
+		return len(message.Data) < 8
+	}
+}
+
+func parseRocketMQRequest() protocol.ParsePkgFn {
+	return func(message *protocol.PayloadMessage) (bool, bool) {
+		// 解析报文内容
+		var (
+			payloadLength int32
+			headerLength  int32
+			serializeType uint8
+		)
+
+		message.ReadInt32(0, &payloadLength)
+		payloadLength = payloadLength + 4
+
+		header := &rocketmqHeader{}
+		//serializeType==0时，序列化类型为JSON，此时直接截取json序列进行反序列化
+		if serializeType = message.Data[4]; serializeType == 0 {
+			message.ReadInt32(4, &headerLength)
+			_, headerBytes := message.ReadJsonString(8, int(headerLength))
+			if err := json.Unmarshal(headerBytes, header); err != nil {
+				return false, true
+			}
+		} else if serializeType == 1 {
+			parseHeader(message, header)
+		} else {
+			return false, true
+		}
+
+		// 通过AddStringAttribute() 或 AttIntAttribute() 存储解析出的属性
+		message.AddIntAttribute(constlabels.RocketMQRequestPayload, int64(payloadLength))
```

Comment:
Payload is the readable payload with fixed length, there is an example with getAsciiString() in dubbo protocol.

Reply:
Go it, thanks for your comments.